core/docker/build.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF 1>&2
+Usage: $0 [-h] [-a <ARCHITECTURES>] [-r <VERSION>]
+Builds the Trino Docker image
+
+-h       Display help
+-a       Build the specified comma-separated architectures, defaults to amd64,arm64
+-r       Build the specified Trino release version, downloads all required artifacts
+EOF
+}
+
+ARCHITECTURES=(amd64 arm64)
+TRINO_VERSION=
+
+while getopts ":a:h:r:" o; do
+    case "${o}" in
+        a)
+            IFS=, read -ra ARCHITECTURES <<< "$OPTARG"
+            ;;
+        r)
+            TRINO_VERSION=${OPTARG}
+            ;;
+        h)
+            usage
+            exit 0
+            ;;
+        *)
+            usage
+            exit 1
+            ;;
+    esac
+done
+shift $((OPTIND - 1))
+
+SOURCE_DIR="../.."
+WORK_DIR="$(mktemp -d)"
+
+# Retrieve the script directory.
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+cd "${SCRIPT_DIR}" || exit 2
+
+if [ -n "$TRINO_VERSION" ]; then
+    for artifactId in io.trino:trino-server:"${TRINO_VERSION}":tar.gz io.trino:trino-cli:"${TRINO_VERSION}":jar:executable; do
+        "${SOURCE_DIR}/mvnw" -C dependency:get -Dtransitive=false -Dartifact="$artifactId"
+    done
+    LOCAL_REPO=$("${SOURCE_DIR}/mvnw" -B help:evaluate -Dexpression=settings.localRepository -q -DforceStdout)
+    cp "$LOCAL_REPO/io/trino/trino-server/${TRINO_VERSION}/trino-server-${TRINO_VERSION}.tar.gz" "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz"
+    cp "$LOCAL_REPO/io/trino/trino-cli/${TRINO_VERSION}/trino-cli-${TRINO_VERSION}-executable.jar" "${WORK_DIR}/trino-cli-${TRINO_VERSION}-executable.jar"
```

Comment:
Instead of doing cp/chmod inside the if maybe you can just set some variables with correct path to those files and have all the `cp` together in one place. Easy to see what the image contains or where things are being added.

Reply:
i.e. something like set `trino_server_path` and `trino_cli_path` in the `if` and `else`. And then do the `cp` where we `cp -R bin ...`.

Reply:
Agreed, and it indeed looks better, thanks!